db/dbformat.h

```diff
@@ -833,6 +902,15 @@ class IterKey {
     key_size_ = 0;
   }
 
+  void ResetSecondaryBuffer() {
+    if (secondary_buf_ != space_for_secondary_buf_) {
+      delete[] secondary_buf_;
+      secondary_buf_ = space_for_secondary_buf_;
+    }
+    secondary_buf_size_ = sizeof(space_for_secondary_buf_);
+    key_size_ = 0;
```

Comment:
Would it make sense to clear `key_size_` iff `key_` points to the secondary buffer?

Reply:
Good catch! This is only supposed to be called when key_ points to secondary buffer, or during destruction. It's good to make a check for this.